lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -243,3 +251,48 @@ def refine_lines_centroid(lines, line_width):
         current += line_length
 
     return KymoLineGroup(interpolated_lines)
+
+
+def refine_lines_gaussian(lines, window=5):
+    """Refine the lines by gaussian peak MLE.
+
+    Parameters
+    ----------
+    lines: List[pylake.KymoLine] or pylake.KymolineGroup
+        Detected traces on a kymograph.
+    window: int
+        Number of pixels on either side of the estimated line to include in the optimization data.
+    """
+    img = lines[0]._image
+    get_window_limits = lambda c: (max(int(c) - window, 0), int(c) + window + 1)
+    pixel_indices = lambda line: zip(line.time_idx.astype(int), np.rint(line.coordinate_idx))
+
+    # check if there are overlapping fit windows
+    window_ranges = np.vstack(
+        [[(t, *get_window_limits(c)) for t, c in pixel_indices(line)] for line in lines]
+    )
+    unique_frames = np.unique(window_ranges[:, 0], return_counts=True)
+    overlap_count = 0
+    for frame, count in zip(*unique_frames):
+        if count > 1:
+            idx = np.argwhere(window_ranges[:, 0] == frame)
+            ranges = window_ranges[idx.squeeze(), 1:]
+            ranges = ranges[np.argsort(ranges[:, 0])]
+            overlap_count += np.sum(ranges[1:, 0] < ranges[:-1, 1])
+    if overlap_count > 0:
+        warnings.warn("there are overlaps")
```

Comment:
So if we go so far as to trigger a warning here, what's the user supposed to do with that information? We continue with the function either way.

Would it be better if the overlapping frames were annotated in the function's returned result instead so that the user could find the problems when further analyzing the data?